### ISSUE TITLE
Confused about Tip 182

### ISSUE THREAD

**Issue by bLueriVerLHR** [QuantlabFinancial/cpp_tip_of_the_week#81]
> ``` cpp
> template<class... Ts, class T>
> void foo(Ts..., T t) { return t; }
> int main() {
>   std::cout << foo(1); // prints 1
>   std::cout << foo(1, 2, 3, 4); // prints 4
>   std::cout << foo(); // compilation error
> }
> ```
> > https://godbolt.org/z/boxjzq

The function foo's return type is void, so std::cout will not get a value from foo.

In the compiler explore, the `foo(1, 2, 3, 4)` also gets an error.

```
<source>:8:16: error: no matching function for call to 'foo'
  std::cout << foo(1, 2, 3, 4); // prints 4
               ^~~
<source>:4:3: note: candidate function [with Ts = <>, T = int] not viable: requires single argument 't', but 4 arguments were provided
T foo(Ts..., T t) { return t; }
  ^
```

**Is this what the tip want?**

I rewrite the function foo like below.
``` cpp
template <class ... Ts, class T>
T foo(T t, Ts ...) { return t; }
```
The second call will print 1 instead of 4.

So I also want to know **how to write a function that looks like the one in tip 182 and gets no error when `prints 4`.**

Maybe I will use function like below:

``` cpp
template <class T>
T foo(T t) { return t;}

template <class ... Ts, class T>
T foo(T t, Ts... args) { return foo(args ...); }
```

**Comment by jgopel** [comment]
I think what you're seeing is a typo in the example. Each tip has an info section, an example section, and a puzzle section. Also, below the puzzle is a solutions section where you can add your solution.

**Comment by krzysztof-jusiak** [comment]
Thanks, fixed the typo - https://github.com/QuantlabFinancial/cpp_tip_of_the_week/commit/09233fa52b048bb1da4999e7fce26d68705f91f4
The example doesn't work in godbolt as the proposal is not implemented and/or accepted ATM. It's based on - https://www.cppstories.com/2021/non-terminal-variadic-args

**Comment by bLueriVerLHR** [comment]
> Thanks, fixed the typo - [09233fa](https://github.com/QuantlabFinancial/cpp_tip_of_the_week/commit/09233fa52b048bb1da4999e7fce26d68705f91f4) The example doesn't work in godbolt as the proposal is not implemented and/or accepted ATM. It's based on - https://www.cppstories.com/2021/non-terminal-variadic-args

Thanks for the link, now I understand. :)
